core/classifiers/CodeClassifier/CodeClassifier.py

```diff
@@ -0,0 +1,695 @@
+        (no. of elements in union of two sets)
+
+    Args:
+        set_a: list(int). First set.
+        set_b: list(int). Second set.
+
+    Returns:
+        float. Jaccard index of two sets.
+
+    """
+    small_set = set_a[:] if len(set_a) < len(set_b) else set_b[:]
+    union_set = set_b[:] if len(set_a) < len(set_b) else set_a[:]
+    for elem in union_set:
+        if elem in small_set:
+            small_set.remove(elem)
+    union_set.extend(small_set)
+
+    if union_set == []:
+        return 0
+
+    small_set = set_a[:] if len(set_a) < len(set_b) else set_b[:]
+    intersection_set = []
+    for elem in small_set:
+        if elem in set_a and elem in set_b:
+            intersection_set.append(elem)
+            set_a.remove(elem)
+            set_b.remove(elem)
+
+    coeff = float(len(intersection_set)) / len(union_set)
+    return coeff
+
+
+def get_program_similarity(fingerprint_a, fingerprint_b):
+    """Find similarity between fingerprint of two programs.
+
+    Args:
+        fingerprint_a: list((int, int)). Fingerprint of first data point.
+        fingerprint_b: list((int, int)). Fingerprint of second data point.
+
+    Returns:
+        float. Similarity between first and second program.
+    """
+    set_a = [h for (h, _) in fingerprint_a]
+    set_b = [h for (h, _) in fingerprint_b]
+    return calc_jaccard_index(set_a, set_b)
+
+
+def generate_top_similars(training_data, top):
+    """Find 'top' nearest neighbours for all programs in dataset.
+
+    Args:
+        training_data: dict. A dictionary containing training training_data.
+            Structure of training_data is as follows,
+                {
+                    ID: {
+                        'source': str. Source code of program.
+                        'class': int. Associated feedback class with program.
+                        'tokens': list(str). Token list of program.
+                        'k_gram_hashes': list(int). K-gram hash values of
+                            program.
+                        'fingerprint': list(int). Extracted fingerprint of
+                            program.
+                    } for each program in training_data.
+                }
+        top: int. No. of nearest neighbours to be identified.
+
+    Returns:
+        dict. Dictionary containing training data with additional attribute
+        'top_similar' in each data point which contains nearest 'top' no. of
+        data points for each data point. Structure of dict is as follows,
+            {
+                ID: {
+                    'source': str. Source code of program.
+                    'class': int. Associated feedback class with program.
+                    'tokens': list(str). Token list of program.
+                    'k_gram_hashes': list(int). K-gram hash values of program.
+                    'fingerprint': list(int). Extracted fingerprint of program.
+                    'top_similar': list(int). Most similar programs to this
```

Comment:
Actually the description _is_ confusing. I would say something like "The indices of the programs that are most similar to this one."